gammapy/estimators/flux_map.py

```diff
@@ -347,6 +347,11 @@ def ts(self):
         self._check_quantity("ts")
         return self._data["ts"]
 
+    @property
+    def ts_scan(self):
+        """ts scan (`Map`)"""
+        return self.stat_scan - np.expand_dims(self.stat.data, 2)
+
     # TODO: always derive sqrt(TS) from TS?
```

Comment:
To rephrase the question, should we as sqrt(TS) as property?
In the case of LC estimation, we adjust only the flux normalisation (with no adjustment of other parameters). So sqrt(TS) is statistically meaningful.
For other flux maps (e.g. from a 3D fit), one can adjust several parameters at the same time, thus sqrt(TS) does not provide a pertinent quantity.
As a conclusion, there is at least one case for which sqrt(TS) is meaningful: one should keep it...

Reply:
This is basically the behaviour we have now. If I understood your comment correctly it would be good to store the number of degrees of freedom somewhere to have a complete solution for the TS -> significance conversion?